plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/CallSiteInline.hs

```diff
@@ -192,8 +192,10 @@ considerInlineSat tm = do
               -- it would have before, but we can make it happen more often.
               -- We could potentially do this safely in non-conservative mode.
               isBindingPure <- isTermBindingPure (varStrictness varInfo) (varDef varInfo)
-              if isBindingPure then
-                pure $ mkApps (varDef varInfo) args
+              if isBindingPure then do
+                -- rename the term before substituting in
+                renamed <- renameTerm (Done (Dupable (varDef varInfo)))
```

Comment:
We should put the `varDef` in as `Done (dupable term)`, that's kind of the point of those types. Just exactly the same way as we do for other things that we put into the substitutions.